integration/test/Test/Spar/STM.hs

```diff
@@ -221,18 +220,19 @@ validateStateLoginAllUsers owner tid state = do
     uid <- bindResponse (createScimUser owner (unScimToken tok) scimUser) $ \resp -> do
       resp.status `shouldMatchInt` 201
       resp.json %. "id" >>= asString
-    when (isNothing mIdp) $ do
-      registerUser OwnDomain tid email
-
-    loginEmail <- case mIdp of
-      Nothing -> scimUser %. "emails" >>= asList >>= assertOne >>= (%. "value") >>= asString
-      Just _ -> pure email
-    maybe (loginWithPassword 200 loginEmail) (loginWithSaml True tid loginEmail) mIdp
-
-    bindResponse (deleteScimUser owner (unScimToken tok) uid) $ \resp -> do
-      resp.status `shouldMatchInt` 204
-
-    maybe (loginWithPassword 403 loginEmail) (loginWithSaml False tid loginEmail) mIdp
+    case mIdp of
+      Nothing -> do
+        registerInvitedUser OwnDomain tid email
+        loginWithPassword 200 email
+        bindResponse (deleteScimUser owner (unScimToken tok) uid) $ \resp -> do
+          resp.status `shouldMatchInt` 204
+        loginWithPassword 403 email
+      Just idp -> do
+        loginEmail <- scimUser %. "emails" >>= asList >>= assertOne >>= (%. "value") >>= asString
```

Comment:
Haven't we mixed up the cases here? :thinking: 

```haskell
    loginEmail <- case mIdp of
      Nothing -> scimUser %. "emails" >>= asList >>= assertOne >>= (%. "value") >>= asString
```

Reply:
I'll fix it